QueryBuilder/Common/Helpers/LinqExpressionParser.cs

```diff
@@ -0,0 +1,315 @@
+        {
+            if (expression is ConstantExpression constantExpression)
+            {
+                if (IsDisplayClass(constantExpression.Value))
+                {
+                    Value = fieldInfo.GetValue(constantExpression.Value);
+                    return;
+                }
+
+                Value = constantExpression.Value;
+                if (Value == null)
+                {
+                    ScalarOperator = ScalarOperators.IS_NULL;
+                }
+            }
+
+            if (expression is UnaryExpression unaryExpression)
+            {
+                HandleConstantExpression(unaryExpression.Operand);
+            }
+
+            if (expression is MemberExpression memberExpression)
+            {
+                HandleMemberExpression(memberExpression);
+            }
+        }
+
+        private void HandleMemberExpression(MemberExpression expression)
+        {
+            if (expression.Member is PropertyInfo propInfo)
+            {
+                if (expression.Expression is MemberExpression me2 && me2.Member is FieldInfo fi && me2.Expression is ConstantExpression ce && IsDisplayClass(ce.Value))
+                {
+                    Value = propInfo.GetValue(fi.GetValue(ce.Value));
+                }
+                else if (string.IsNullOrWhiteSpace(PropertyName))
+                {
+                    PropertyName = GetPropertyName(expression);
+                }
+            }
+
+            if (expression.Member is FieldInfo fieldInfo)
+            {
+                HandleConstantExpression(expression.Expression, fieldInfo);
+            }
+        }
+
+        private static bool IsDisplayClass(object o)
+        {
+            if (o is null)
+            {
+                return false;
+            }
+
+            return o.GetType().Name.Contains("DisplayClass");
+        }
+
+        private string GetPropertyName(MemberExpression expression)
+        {
+            var propInfo = expression.Member as PropertyInfo;
+            var attribute = propInfo.GetPropertyAttributeValueL<JsonPropertyNameAttribute, string>(attr => attr.Name);
+            var propName = string.IsNullOrEmpty(attribute)
+                    ? string.Equals(propInfo.Name, DTID, StringComparison.OrdinalIgnoreCase) ? ADTDTID : propInfo.Name.ToLowerFirstChar()
+                    : attribute;
+
+            // multi level property
+            if (expression.Expression is MemberExpression me2)
+            {
+                var parentPropInfo = me2.Member as PropertyInfo;
+                var parentAttribute = parentPropInfo.GetPropertyAttributeValueL<JsonPropertyNameAttribute, string>(attr => attr.Name);
+                var parentPropName = string.IsNullOrEmpty(parentAttribute)
+                        ? string.Equals(propInfo.Name, DTID, StringComparison.OrdinalIgnoreCase) ? ADTDTID : propInfo.Name.ToLowerFirstChar()
+                        : parentAttribute;
```

Comment:
A little duplication here of lines 203-206 above. I think this code is novel enough to be extracted into a separate method.